src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/ILLink/ILLinkTrim.xml

```diff
@@ -2,4 +2,12 @@
     <assembly fullname="System.Private.Runtime.InteropServices.JavaScript">
         <type fullname="System.Runtime.InteropServices.JavaScript.Runtime" />
     </assembly>
+    <!--
+    Preserve System.Delegate.DynamicInvoke which is called from javascript
```

Comment:
Is there any place in IL that can use `[DynamicDependency]` instead?

Another thought is to put a new method on `System.Runtime.InteropServices.JavaScript.Runtime` that JS invokes into to call the `Delegate.DynamicInvoke` method. The whole `Runtime` class is preserved, so nothing in it will be trimmed.

cc @vitek-karas

Reply:
Thanks for the suggestions but this should all be resolved by calling Invoke on the Delegate as suggested above all.